pkg/cleaner/azure/cleaner.go

```diff
@@ -55,39 +64,59 @@ func NewCleaner(config CleanerConfig) (*Cleaner, error) {
 }
 
 func (c *Cleaner) Clean(ctx context.Context) error {
-	// TODO cleanup peeringso
-	//
-	//     https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network#VirtualNetworksClient
-	//     https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network#VirtualNetworkPeeringsClient
-	//
-
-	r, err := c.virtualNetworksClient.ListAll(ctx)
-	if err != nil {
-		return microerror.Mask(err)
-	}
+	c.logger.LogCtx(ctx, "level", "debug", "message", "starting Azure CI cleanup")
+
+	for _, i := range c.installations {
+		r, err := c.virtualNetworksClient.List(ctx, i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
 
-	for _, v := range r.Values() {
-		// We skip all virtual networks of installations we do not want to cleanup.
-		if !containsString(c.installations, *v.Name) {
-			continue
+		for {
+			for _, v := range r.Values() {
+				for _, p := range *v.VirtualNetworkPeerings {
+					if !isCIResource(*p.Name) {
+						continue
+					}
+
+					_, err = c.groupsClient.Get(ctx, *p.Name)
+					if IsResourceGroupNotFound(err) && p.PeeringState == network.VirtualNetworkPeeringStateDisconnected {
+						c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting vnet peering '%s'", *p.Name))
+
+						_, err := c.virtualNetworkPeeringsClient.Delete(ctx, i, *v.Name, *p.Name)
+						if err != nil {
+							return microerror.Mask(err)
+						}
+
+						c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted vnet peering '%s'", *p.Name))
+
+						time.Sleep(1 * time.Second)
+						continue
+					} else if err != nil {
+						return microerror.Mask(err)
+					}
+				}
+			}
+
+			if r.NotDone() {
+				err = r.Next()
+				if err != nil {
+					return microerror.Mask(err)
+				}
+				continue
+			}
```

Comment:
This is not obvious but can be simplified: https://github.com/giantswarm/azure-operator/blob/4cdddedde96f5c2510ecf10455395beef65f6b12/service/controller/v1/resource/endpoints/nic.go#L34

Reply:
Yeah but then you ignore the error and I am against this in general.